src/Styleguide/Components/FollowArtistPopover/FollowArtistPopoverRow.tsx

```diff
@@ -0,0 +1,83 @@
+import { Image } from "@artsy/palette"
+import { FollowArtistPopoverRow_artist } from "__generated__/FollowArtistPopoverRow_artist.graphql"
+import { FollowArtistPopoverRowMutation } from "__generated__/FollowArtistPopoverRowMutation.graphql"
+import { ContextProps } from "Artsy"
+import React from "react"
+import { commitMutation, createFragmentContainer, graphql } from "react-relay"
+import { RecordSourceSelectorProxy, SelectorData } from "relay-runtime"
+
+interface Props extends ContextProps {
+  artist: FollowArtistPopoverRow_artist
+}
+
+interface State {
+  swappedArtist: FollowArtistPopoverRow_artist
+}
+
+class FollowArtistPopoverRow extends React.Component<Props, State> {
+  state = {
+    swappedArtist: null,
+  }
+  handleClick(artistID: string, relayID: string) {
+    const { user, relay } = this.props
+    if (user && user.id) {
+      commitMutation<FollowArtistPopoverRowMutation>(relay.environment, {
+        mutation: graphql`
+          mutation FollowArtistPopoverRowMutation($input: FollowArtistInput!) {
+            followArtist(input: $input) {
+              artist {
+                __id
+                related {
+                  suggested(first: 1, exclude_followed_artists: true) {
+                    edges {
+                      node {
+                        __id
+                        ...FollowArtistPopoverRow_artist @relay(mask: false)
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        `,
+        variables: {
+          input: { artist_id: artistID, unfollow: false },
+        },
+        updater: (store: RecordSourceSelectorProxy, data: SelectorData) => {
+          const newEdge = data.followArtist.artist.related.suggested.edges[0]
+          this.setState({ swappedArtist: newEdge.node })
```

Comment:
This component renders a row for a given artist, which can either be the artist it started out with, or one that gets swapped in as you click + follow (which is set in state).

At first, I was trying to update the Relay store to remove the clicked on artist and replace it with the new one, but that seems like overkill/unnecessary when this `Row` component can just use its own state for that.